src/panels/lovelace/editor/config-elements/hui-generic-entity-row-editor.ts

```diff
@@ -45,127 +38,107 @@ export class HuiGenericEntityRowEditor
-          ></paper-input>
-          <ha-icon-picker
-            .label=${this.hass!.localize(
-              "ui.panel.lovelace.editor.card.generic.icon"
-            )}
-            .value=${this._config.icon}
-            .configValue=${"icon"}
-            .placeholder=${entityState?.attributes.icon}
-            .fallbackPath=${!this._icon &&
-            !entityState?.attributes.icon &&
-            entityState
-              ? domainIcon(computeDomain(entityState.entity_id), entityState)
-              : undefined}
-            @value-changed=${this._valueChanged}
-          ></ha-icon-picker>
-        </div>
-        <ha-select
-          label="Secondary Info"
-          .configValue=${"secondary_info"}
-          @selected=${this._valueChanged}
-          @closed=${stopPropagation}
-          .value=${this._config.secondary_info || "none"}
-          naturalMenuWidth
-          fixedMenuPosition
-        >
-          <mwc-list-item value=""
-            >${this.hass!.localize(
-              "ui.panel.lovelace.editor.card.entities.secondary_info_values.none"
-            )}</mwc-list-item
-          >
-          ${Object.keys(SecondaryInfoValues).map((info) => {
-            if (
-              !("domains" in SecondaryInfoValues[info]) ||
-              ("domains" in SecondaryInfoValues[info] &&
-                SecondaryInfoValues[info].domains!.includes(domain))
-            ) {
-              return html`
-                <mwc-list-item .value=${info}>
-                  ${this.hass!.localize(
-                    `ui.panel.lovelace.editor.card.entities.secondary_info_values.${info}`
-                  )}
-                </mwc-list-item>
-              `;
-            }
-            return "";
-          })}
-        </ha-select>
-      </div>
+      <ha-form
+        .hass=${this.hass}
+        .data=${this._config}
+        .schema=${schema}
+        .computeLabel=${this._computeLabelCallback}
+        @value-changed=${this._valueChanged}
+      ></ha-form>
     `;
   }
 
-  private _valueChanged(ev: EntitiesEditorEvent): void {
-    if (!this._config || !this.hass) {
-      return;
-    }
-    const target = ev.target! as EditorTarget;
-    const value = target.value || ev.detail?.item?.value;
-
-    if (this[`_${target.configValue}`] === value) {
-      return;
-    }
+  private _valueChanged(ev: CustomEvent): void {
+    fireEvent(this, "config-changed", { config: ev.detail.value });
+  }
 
-    if (target.configValue) {
-      if (value === "" || !value) {
-        this._config = { ...this._config };
-        delete this._config[target.configValue!];
-      } else {
-        this._config = {
-          ...this._config,
-          [target.configValue!]: value,
-        };
-      }
+  private _computeLabelCallback = (schema: HaFormSchema) => {
+    if (schema.name === "entity") {
+      return `${this.hass!.localize(
+        "ui.panel.lovelace.editor.card.generic.entity"
+      )} (${this.hass!.localize(
```

Comment:
I don't think that we should add required to the label like this. I think that selectors just need to render an `*` if something is required. (and so we do that by setting `required: true`,  not in label computation)